### ISSUE TITLE
feat: format .d.ts files

### ISSUE THREAD

**Issue by KnorpelSenf** [denoland/vscode_deno#780]
**Is your feature request related to a problem? Please describe.**

For some libraries I work with `.d.ts` files. Many related projects are written in Deno, so I'd like to use the Deno formatter for the `.d.ts` files, too. While I can type `deno fmt` in the terminal to format the declaration files, the VSCode extension for some reason does not register as a plugin that is able to format the files.

**Describe the solution you'd like**

Configure this extension as the default formatter for declaration files, and then be able to use it to format the files.

**Comment by Leokuma** [comment]
Can you provide the steps to reproduce the problem? I was able to format a `.d.ts` file with the Deno extension.

**Comment by KnorpelSenf** [comment]
1. Open a .d.ts file
2. Press Ctrl+Shift+I
3. See an error message that no formatter is installed for .d.ts files

I'll try to play around with my setup, perhaps some combination with other extensions breaks this?

**Comment by KnorpelSenf** [comment]
Right, I'm also using whatever is the default on my machine, but I see that these could differ. Shift+Alt+F doesn't do anything for me. Doing Ctrl+Shift+P and formatting the code via that panel used to reproduce the issue, too.

I didn't find time to play around further with this, will get back to you then.

**Comment by KnorpelSenf** [comment]
This is the error I get: ![image](https://user-images.githubusercontent.com/12952387/212705678-52bdd933-519f-4c98-baeb-33586fa7dd10.png)

I now also found out why the formatter isn't available. I am only allowed to use it if the Deno extension is enabled for the respective project. This behaviour is understandable for many other LSP features, but IMO formatting in particular should be an exception here.

**Comment by iwoplaza** [comment]
Behaves the same for me. I would expect formatting to be available even though the Deno LSP is disabled for a project. According to #657, this was a possibility in 2022, but I was not able to achieve the same behavior nowadays.

**Comment by ayonli** [comment]
Hey guys, I just couldn't wait, so I built another VSCode extension instead (Thanks to the help of AI). Check it out:

https://marketplace.visualstudio.com/items?itemName=ayonli.deno-tools

This extension allows us to use Deno formatter and linter without the Deno LSP.

**Comment by KnorpelSenf** [comment]
@ayonli why not fix up this extension instead?

**Comment by ayonli** [comment]
@KnorpelSenf Because I’m not familiar with this extension, nor do I have any experience of writing a VSCode extension.

My first thought is the same, to provide a PR to this extension. But after reviewing some code of the project, I still have problem figuring out how deeply the linter is integrated into the LSP. So naturally I took another approach.

I know the ‘deno lint’ CLI has a ‘--json’ option that prints structured diagnostics, and as my first try to writing a VSCode extension and vibe coding (I’ve used auto-completion for a very long time, but never trusted vibe coding, however my colleagues have been suggesting me for quite a while), I decided to write another extension instead.

The new extension serves me three purposes:

1. Had some experience of writing a VSCode extension, by reviewing what the AI write and the API used and providing feedback to the AI generated code.
2. Gain experience of vibe coding and build trust in AI, which is an inevitable future anyway.
3. Quickly support my workflow of having an VSCode extension that allows me to use Deno tools in existing Node.js and frontend projects.

I may be merge the functionalities into the Deno extension or the Deno team catches up some time in the future and make the new extension unnecessary, but for now, it has done a great job for me.
